hermes/apps/athena/modules/rbac-registration/src/lib.rs

```diff
@@ -71,97 +59,55 @@ impl exports::hermes::init::event::Guest for RbacRegistrationComponent {
-    let (active_1, inactive_1) = get_active_inactive_stake_address(
-        rbac_1.stake_addresses(),
-        rbac_1.catalyst_id(),
-        persistent,
-        volatile,
-        network,
-        network_resource,
-    )
-    .unwrap();
-    log_info(
-        file!(),
-        FUNCTION_NAME,
-        "",
-        &format!(
-            "📕 From catalyst id {cat_id_1}: Cat ID {}, All stake addresses: {:?}, Active stake address: {active_1:?}, Inactive stake address: {inactive_1:?})",
-            rbac_1.catalyst_id(),
-            rbac_1.stake_addresses()
-        ),
-        None,
-    );
-
-    /* cspell:disable */
-    // Testing get rbac data from stake address
-    // `stake_test1urgduxg0zec4zw4k3v33ftsc79ffdwzzj6ka2d3w86dyudqmmj5tv`
-    // `e0d0de190f1671513ab68b2314ae18f15296b84296add5362e3e9a4e34`
-    // This stake address is taken by
-    // `preprod.cardano/ZtnkJZNZHskfS6mhChVstXRrhDPUdzTGwFidSg_YjsA`
-    /* cspell:enable */
-    let hash: Hash<28> = "d0de190f1671513ab68b2314ae18f15296b84296add5362e3e9a4e34"
-        .parse()
-        .unwrap();
-    let stake_address = StakeAddress::new(Network::Preprod, false, hash.into());
-
-    let rbac_2 = get_rbac_chain_from_stake_address(
-        persistent,
-        volatile,
-        stake_address.clone(),
-        network,
-        network_resource,
-    )
-    .unwrap()
-    .unwrap();
-    // Active 1, No inactive
-    let (active_2, inactive_2) = get_active_inactive_stake_address(
-        rbac_2.stake_addresses(),
-        rbac_2.catalyst_id(),
-        persistent,
-        volatile,
-        network,
-        network_resource,
-    )
-    .unwrap();
-    log_info(
-        file!(),
-        FUNCTION_NAME,
-        "",
-        &format!(
-            "📕 From stake address {stake_address}: Cat ID {}, All stake addresses: {:?}, Active stake address: {active_2:?}, Inactive stake address: {inactive_2:?})",
-            rbac_2.catalyst_id(),
-            rbac_2.stake_addresses()
-        ),
-        None,
-    );
+impl exports::hermes::http_gateway::event::Guest for RbacRegistrationComponent {
+    fn reply(
+        _body: Vec<u8>,
+        _headers: Headers,
+        path: String,
+        _method: String,
+    ) -> Option<HttpGatewayResponse> {
+        let network = cardano::api::CardanoNetwork::Preprod;
+        let lookup = parse_query_param(&path, "lookup");
+        let result = endpoint_v1(lookup, network);
+        let code = result.status_code();
+
+        Some(HttpGatewayResponse::Http(HttpResponse {
+            code,
+            headers: vec![("content-type".to_string(), vec![
+                "application/json".to_string()
+            ])],
+            body: Bstr::from(match result.to_json() {
+                Ok(json) => json,
+                Err(e) => format!("{{\"error\": \"Failed to serialize response: {}\"}}", e),
+            }),
+        }))
+    }
+}
+
+/// Extract query parameter from path.
+// Temporary way to get the query parameter.
```

Comment:
Can we replace this with the `parse` function from `url` crate?
See here: https://docs.rs/url/2.5.7/url/